tests/markov/hmm/test_output_models.py

```diff
@@ -80,7 +80,9 @@ def test_sample(self):
             np.array([1] * 30000 + [2] * 70000)  # state 1
         ]
         m.sample(obs_per_state)
-        np.testing.assert_array_almost_equal(m.output_probabilities, np.array([[.5, .5, 0.], [0, .3, .7]]), decimal=2)
+        # the output probabilities of the unpopulated states are left as-is (can't sample), hence we compare against
+        # [[.5, .5, .1], [.1, .3, .7]] instead of [[.5, .5, .0], [.0, .3, .7]]
+        np.testing.assert_array_almost_equal(m.output_probabilities, np.array([[.5, .5, .1], [.1, .3, .7]]), decimal=2)
```

Comment:
is this what will happen in "real" cases too? is this what we want?

Reply:
I was actually wondering the same thing, this is replicating the bhmm behavior. It does happen if some states are not visited in a certain trajectory. One has to make sure afterwards that the output probabilities are still row-stochastic (0.5 + 0.5 + 0.1 is clearly not 1). I believe this is built in because the discrete output model in a HMM samples from the dirichlet distribution which can't be sampled if there are no counts in a state so they probabilities are just left as-is.

Reply:
so what is the "correct" fix? normalizing [0.5, 0.5, 0.1] on account of some kind of prior model? or knowing that the 0.1 is "meaningless" or preserved from the original/unsampled? would it make sense to give a warning or printout? of like certain states are unsampled? or return an object/perhaps mask somehow indicating unsampled states?

Reply:
leaving it as it is for now, lets discuss later